app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioGroup.kt

```diff
@@ -0,0 +1,905 @@
+            .weight(1f)
+            .width(Min)
+            .padding(itemPadding),
+        )
+      }
+    }
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+private fun HorizontalRadioGroupWithLabel(
+  data: List<OptionData>,
+  groupLabelText: String,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  contentPaddingValues: PaddingValues,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  val shape = radioGroupSize.toOptionSize().size(LeftAligned).shape
+  Surface(
+    shape = shape,
+    color = radioOptionColors.containerColor,
+    modifier = modifier,
+  ) {
+    Column(
+      Modifier
+        .padding(
+          contentPaddingValues,
+        ),
+    ) {
+      val labelTextColor = radioOptionColors.labelTextColor(groupLockedState)
+      HedvigText(
+        text = groupLabelText,
+        style = calculateHorizontalGroupLabelTextStyle(radioGroupSize),
+        color = labelTextColor,
+      )
+      Spacer(Modifier.height(16.dp))
+      FlowRow(
+        modifier = Modifier.fillMaxWidth(),
+        maxItemsInEachRow = 2,
+        overflow = FlowRowOverflow.Visible,
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        horizontalArrangement = Arrangement.spacedBy(16.dp),
+      ) {
+        for (option in data) {
+          val optionTextColor = radioOptionColors.optionTextColor(option.lockedState)
+          val itemModifier = if (data.size == 2) {
+            Modifier.width(Min)
+          } else {
+            Modifier
+              .weight(1f)
+              .width(Min)
+          }
+          val lockedState = calculateLockedStateForItemInGroup(option, groupLockedState)
+          HorizontalWithLabelRadioOption(
+            enabled = lockedState == NotLocked,
+            style = calculateHorizontalGroupOptionTextStyle(radioGroupSize),
+            textColor = optionTextColor,
+            modifier = itemModifier
+              .align(Alignment.CenterVertically),
+            optionText = option.optionText,
+            onClick = {
+              onOptionClick(option.id)
+            },
+            chosenState = option.chosenState,
+            lockedState = lockedState,
+          )
+        }
+      }
+    }
+  }
+}
+
+@Composable
+private fun HorizontalWithLabelRadioOption(
+  optionText: String,
+  onClick: () -> Unit,
+  lockedState: LockedState,
+  chosenState: ChosenState,
+  enabled: Boolean,
+  style: TextStyle,
+  textColor: Color,
+  modifier: Modifier = Modifier,
+) {
+  val interactionSource = remember { MutableInteractionSource() }
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier
+      .clickable(
+        enabled = enabled,
+        role = Role.RadioButton,
+        interactionSource = interactionSource,
+        indication = ripple(
+          bounded = false,
+          radius = 50.dp,
+          // todo: pls check out this ripple. I like this one, but maybe it's too much
```

Comment:
We can take a look at this within some test session etc. to see what others think too